src/int.rs

```diff
@@ -21,6 +21,8 @@ pub trait PrimInt
     + CheckedSub<Output=Self>
     + CheckedMul<Output=Self>
     + CheckedDiv<Output=Self>
+    + CheckedShl<usize, Output=Self>
+    + CheckedShr<usize, Output=Self>
```

Comment:
Unfortunately, this is a breaking change.  We don't control all implementations of `PrimInt`, so adding new constraints would break others, however theoretical such others may be.